nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/molecule/CopyMoleculeEventHandler.java

```diff
@@ -0,0 +1,93 @@
+package org.nmrfx.analyst.gui.molecule;
+
+import javafx.event.EventHandler;
+import javafx.scene.input.Clipboard;
+import javafx.scene.input.DataFormat;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyCodeCombination;
+import javafx.scene.input.KeyCombination;
+import javafx.scene.input.KeyEvent;
+import org.nmrfx.chemistry.io.MoleculeIOException;
+import org.nmrfx.chemistry.io.SDFile;
+import org.nmrfx.processor.gui.FXMLController;
+import org.nmrfx.processor.gui.PolyChart;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+public class CopyMoleculeEventHandler implements EventHandler<KeyEvent> {
+    private static final Logger log = LoggerFactory.getLogger(CopyMoleculeEventHandler.class);
+    private final KeyCodeCombination keyCodeCombination = new KeyCodeCombination(KeyCode.V, KeyCombination.SHORTCUT_DOWN);
+
+    /**
+     * Checks for paste (shortcut + v) KeyEvent and delete KeyEvents.Paste will attempt
+     * to add a molecule to the canvas from the clipboard. Delete will remove a molecule from the
+     * canvas if it exists.
+     * @param event the event which occurred
+     */
+    @Override
+    public void handle(KeyEvent event) {
+        if (keyCodeCombination.match(event)) {
+            // Don't try to paste the molecule if there is no dataset displayed
+            PolyChart activeChart = FXMLController.getActiveController().getActiveChart();
+            if (activeChart.getDataset() == null) {
+                return;
+            }
+            String molContents = parseClipboard();
+            // Not a mol file without multiple lines
+            if (molContents.indexOf('\n') < 0) {
+                return;
+            }
+            // Use the first line of the string as the filename
+            String filename = molContents.split("\n")[0];
+            try {
+                SDFile.read(filename, molContents);
+            } catch (MoleculeIOException e) {
+                log.error("Unable to read molecule file.");
+                return;
+            }
+            MoleculeUtils.addActiveMoleculeToCanvas();
+        } else if (event.getCode() == KeyCode.DELETE) {
+            MoleculeUtils.removeMoleculeFromCanvas();
```

Comment:
Added a delete option in case someone changed their mind about the molecule they pasted, and there is no remove molecule option in the menu bar if the molecule is pasted into a 2D dataset or into an FID.